### PR TITLE
Smaller docs updates.

### LINKED ISSUE #309: Documentation issues (just add to it when you notice something)
collecting place for documentation corrections, each as a comment.

### PR DESCRIPTION
**Pull request by rsmmr**
Closes #309.

### CODE REVIEW

File: doc/zeek.rst [hunk 1]
```diff
@@ -428,6 +436,32 @@ processing:
...
+   analyzer through the ``enable "<name>"`` keyword. The name of the
+   analyzer comes out of the EVT file: it is the ``ANALYZER_NAME``
+   with the double colons replaced with an underscore (e.g.,
+   ``spicy::HTTP`` turns into ``enable "spicy_HTTP"``.
+
+2. You should call ``zeek::confirm_protocol()`` (see
+   :ref:`zeek_functions`) from a hook inside your grammar at a point
+   when the parser can be reasonably certain that it is processing the
+   expected protocol. Optionally, you may also call
+   ``zeek::reject_protocol()`` when you're sure the parser is *not*
```
Comment (bbannier) [inline doc/zeek.rst:461]: Let's call out where one would call these, e.g., `confirm_protocol` in a `%done` hook, and `reject_protocol` in an `%error` hook.

File: tests/zeek/evt/ssh-banner.zeek [hunk 2]
```diff
@@ -30,10 +30,20 @@ public type Banner = unit {
...
+    enable "spicy_SSH"
+    tcp-state responder
+}
+# @TEST-END-FILE
+
 # @TEST-START-FILE ssh.evt
 protocol analyzer spicy::SSH over TCP:
-    parse with SSH::Banner,
-    port 22/tcp;
+    # no port, we're using the signature
```
Comment (bbannier) [inline tests/zeek/evt/ssh-banner.zeek:45]: Can we also add a `reject_protocol` case to this test?

File: doc/programming/parsing.rst [hunk 3]
```diff
@@ -397,7 +397,7 @@ There are three location where hooks can be implemented:
    When a hook executes, it has access to the current unit instance
    through the ``self`` identifier. The state of that instance will
    reflect where parsing is at that time. In particular, any field
-   that haven't been parsed yet, will remain unset. (You can use the
+   that hasn't been parsed yet, will remain unset. (You can use the
```
Comment (bbannier) [inline doc/programming/parsing.rst:400]:
```suggestion
   that hasn't been parsed, yet will remain unset. (You can use the
```

File: doc/zeek.rst (hunk 1 above)
Comment (rsmmr) [reply doc/zeek.rst:461]: I'll add a note on `reject_protocol`. For confirming, there's not much of a common case, depends on there protocol and could be in the middle of some unit.

File: doc/programming/parsing.rst (hunk 3 above)
Comment (rsmmr) [reply doc/programming/parsing.rst:400]: actually the original version describes it better I think

File: doc/zeek.rst (hunk 1 above)
Comment (rsmmr) [reply doc/zeek.rst:461]: Good point, will do.

**Pull request merged**